dist/radar_client.js

```diff
@@ -156,235 +393,200 @@ Client.prototype.currentClientId = function() {
-    // options is an optional argument
-    if (typeof options == 'function') {
-      callback = options;
-    } else {
-      message.options = options;
-    }
-    // Sync v1 for presence scopes acts inconsistently. The result should be a
-    // "get" message, but it is actually a "online" message.
-    // So force v2 and translate the result to v1 format.
-    if (propertyName == 'sync' && !message.options && scope.match(/^presence.+/)) {
-      message.options = { version: 2 };
-      this.when('get', function(message) {
-        var value = {}, userId;
-        if (!message || !message.to || message.to != scope) {
-          return false;
-        }
+var callbackSet = function (callback, options) {
+  if (typeof options === 'function') { callback = options; }
+  return callback;
+};
 
-        for (userId in message.value) {
-          if (message.value.hasOwnProperty(userId)) {
-            // Skip when not defined; causes exception in FF for 'Work Offline'
-            if (!message.value[userId]) { continue; }
-            value[userId] = message.value[userId].userType;
-          }
-        }
-        message.value = value;
-        message.op = 'online';
-        if (callback) {
-          callback(message);
-        }
-        return true;
-      });
-    } else {
-      this.when('get', function(message) {
-        if (!message || !message.to || message.to != scope) {
-          return false;
-        }
-        if (callback) {
-          callback(message);
-        }
-        return true;
-      });
+// sync returns the actual value of the operation
+Client.prototype.sync = function (scope, options, callback) {
+  var request = Request.buildSync(scope, options);
+
+  var whenCallback = function (response) {
+    if (response && response.getAttr('to') === request.getAttr('to')) {
+      if (request.getVersion() === 1 && request.isPresence()) {
+        response.forceV2Presence();
+      }
+      if (callback) {
+        callback(response.getMessage());
+      }
+      return true;
     }
-    // sync/get never register or return acks (since they always send back a
-    // data message)
-    return this._write(message);
+    return false;
   };
+
+  callback = callbackSet(callback, options);
+  this.when('get', whenCallback);
+
+  // sync does not return ACK (it sends back a data message)
+  return this._write(request);
 };
 
-var props = ['get', 'sync'];
-for(var i = 0; i < props.length; i++){
-  init(props[i]);
-}
+// get returns the actual value of the operation
+Client.prototype.get = function (scope, options, callback) {
+  var request = Request.buildGet(scope, options);
+
+  var whenCallback = function (response) {
+    if (response && response.getAttr('to') === request.getAttr('to')) {
+      if (callback) {
+        callback(response.getMessage());
+      }
+      return true;
+    }
+    return false;
+  };
+
+  callback = callbackSet(callback, options);
+  this.when('get', whenCallback);
```

Comment:
Actually, let's not make it a variable. It's breaking the flow of reading the code. Expand the function here. It will look much better I think. Especially, with 'function(response) {'